src/java.base/share/classes/sun/security/pkcs/SigningCertificateInfo.java

```diff
@@ -132,7 +128,7 @@ class ESSCertId {
 
     private static volatile HexDumpEncoder hexDumper;
 
-    private byte[] certHash;
+    private final byte[] certHash;
     private GeneralNames issuer;
     private SerialNumber serialNumber;
```

Comment:
You could also mark these as `final` and add an `else` block to the ctor which sets them to `null`.

Reply:
Fixed as suggested.